### DISCUSSION TITLE
The model does not have an ir_version set properly.

### DISCUSSION THREAD

**Discussion by ns-ofekshochat** [onnx/onnx#5424] [Q&A]
I have exported a pytorch model (the gpt2 one from the transformers package) with `torch.onnx.export` . I've loaded the model and the checker complained that `The model does not have an ir_version set properly.`
I first tried `opset_version` parameter, and later set it directly with `model.ir_version = onnx.IR_VERSION` but to no avail. 
has anyone encountered this before?

**Comment by xadupre** [comment]
Did you try to use the paramter `opset_version=` when calling `torch.onnx.export`?

**Reply by ns-ofekshochat** [reply]
I did, yes

**Reply by xadupre** [reply]
So you got the right opset but the wrong ir_version? It seems a bug of the converter then. onnxruntime only supports opset 18 and ir_version 8 right now.

**Reply by ns-ofekshochat** [reply]
does seem so. well, that doesn't seem fun

**Reply by xadupre** [reply]
+ @thiagocrepaldi who worked on torch.onnx.

**Comment by justinchuby** [comment]
Please also share the versions of the packages you are using: onnx, torch, onnxruntime etc.

**Reply by ns-ofekshochat** [reply]
torch 2.1.0.dev20230714+cu121
onnx 1.14.0
onnxruntime 1.15.1

**Reply by justinchuby** [reply]
What’s the IR version being used?

**Comment by armwaheed** [comment]
@ns-ofekshochat, here's the fix that worked for me:
* Update your model.onnx header with a valid `model.ir_version`:
   ```
   import onnx
   
   model = onnx.load("/path/to/your/model.onnx")
   model.ir_version = 9
   onnx.save(model, "/path/to/your/model.onnx")
   ```
* It's frustrating but you need to run this code entirely separately and then work with the `model.onnx` file that was saved, with the updated header information

**Reply by justinchuby** [reply]
Thanks for the answer - You can consider setting `load_external_data=False` in `onnx.load` to ensure the model is not altered if it has reference to external tensor files.
